src/transformers/generation/flax_utils.py

```diff
@@ -581,6 +587,22 @@ def _greedy_search(
         # per batch-item holding current token in loop.
         sequences = jnp.full((batch_size, max_length), pad_token_id, dtype=jnp.int32)
         sequences = lax.dynamic_update_slice(sequences, input_ids, (0, 0))
+        if hasattr(self.config, "vocab_size") and self.config.vocab_size is not None:
+            vocab_size = self.config.vocab_size
+        elif (
+            hasattr(self.config, "decoder")
+            and hasattr(self.config.decoder, "vocab_size")
+            and self.config.decoder.vocab_size is not None
+        ):
+            vocab_size = self.config.decoder.vocab_size
+        elif (
+            hasattr(self.config, "encoder")
+            and hasattr(self.config.encoder, "vocab_size")
+            and self.config.encoder.vocab_size is not None
+        ):
+            vocab_size = self.config.encoder.vocab_size
+
+        scores = jnp.float16(jnp.ones((batch_size, max_length, vocab_size)) * np.array(-1.0e7))
```

Comment:
This shouldn't be float16:
```suggestion
        scores = jnp.ones((batch_size, max_length, vocab_size)) * np.array(-1.0e7)
```

Reply:
Updated

Reply:
After removing the conversion to float16, something maybe not important occured to the resulted score values: scores for the first token converted from `-inf` to `-10000000.` but scores of the next tokens remain unchanged.

Reply:
This should be fine! Note that we can't assume `float16` since we need the code to be both CPU and TPU compatible (which use `float32` and `bfloat16` respectively)

Reply:
We do this with beam  search with no issues in general: https://github.com/huggingface/transformers/blob/d0b5002378daabf62769159add3e7d66d3f83c3b/src/transformers/generation/flax_utils.py#L813